src/python/pants/backend/jvm/tasks/jar_task.py

```diff
@@ -445,6 +442,9 @@ def add_target(self, target, recursive=False):
         elif agents:
           self._add_agent_manifest(agents[0], self._manifest)
           products_added = True
+      elif isinstance(target, JavaAgent):
+        self._add_agent_manifest(target, self._manifest)
+        products_added = True
```

Comment:
I think this'd be a little easier to follow if there were a comment above the conditional block here pointing out that it is responsible for adding an agent if it exists.

Reply:
Good call :) Done